### PR TITLE
fix tests for ci

### PR DESCRIPTION
**Pull request by generalhenry**
rebased version of https://github.com/appcelerator/amp/pull/8

### CODE REVIEW

File: docker-compose.test.yml [hunk 1]
```diff
@@ -4,9 +4,16 @@ services:
...
-      ES_URL: http://elasticsearch:9200
-  elasticsearch:
-    image: appcelerator/elasticsearch-amp:latest
+    command: ["go", "test", "-v", "github.com/appcelerator/amp/data/etcd"]
+  etcd:
+    image: quay.io/coreos/etcd
+    command: [
+      "etcd",
+      "-name", "etcd0",
+      "-advertise-client-urls", "http://etcd:2379,http://etcd:4001",
```
Comment (subfuzion) [inline docker-compose.test.yml]: Port `4001` not needed. 2379 and 2380 are the IANA assigned ports now (http://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml?search=etcd).

File: data/etcd/store_test.go [hunk 2]
```diff
@@ -15,9 +15,9 @@ const (
 )
 
 var (
-	config server.Config = server.Config{
+	config = server.Config{
 		Port:          ":50101",
-		EtcdEndpoints: []string{"http://localhost:2379"},
+		EtcdEndpoints: []string{"http://etcd:2379"},
```
Comment (subfuzion) [inline data/etcd/store_test.go]: As much as I want CI tests to pass, I hate to break local tests which we use during development, and I don't want it to be error prone changing this and then remember to change it back. I suppose a quick and dirty way to fix for local development right now might be to add an entry to the HOSTS file that maps `etcd` to 127.0.0.1. I don't see any other easy ways besides running the tests themselves from a container attached to the swarm, but that involves a somewhat tedious build step. Thoughts?

Comment (generalhenry) [inline data/etcd/store_test.go]: I'm adding a command line flag

Comment (generalhenry) [inline data/etcd/store_test.go]: I added an env for setting it, defaulting to localhost

**Comment by generalhenry** [comment]
I added an env to set the etcd hosts so local and ci tests flows are preserved

File: docker-compose.test.yml [hunk 3]
```diff
@@ -4,9 +4,16 @@ services:
...
+      "-listen-client-urls", "http://0.0.0.0:2379,http://0.0.0.0:4001",
```
Comment (subfuzion) [inline docker-compose.test.yml]: Same: just use 2379.

**Comment by subfuzion** [comment]
LGTM. It's really nice to see passing CI builds now. Thanks! 👍

**Pull request merged**